### PR TITLE
Changes shoes a bit so that clown shoes make a sound every step (except advanced ones)

### PR DESCRIPTION
**Pull request by boy2mantwicethefam**
Stepstaken is removed because it's not used by anything
Replaced with step_sound_counter and step_sound_threshold, the former keeps track of the steps made and gets reset when a sound happens and the latter is how many steps required until sound is played
Clown shoes now make a noise every step except for the advanced ones, but you can var-edit the advanced ones to 1 if you're insidious
:cl:
 * tweak: Clown shoes now make a noise every step. This does not apply to the advanced clown shoes.

### CODE REVIEW

**Comment by SonixApache** [comment]
*why*

**Comment by boy2mantwicethefam** [comment]
someone suggested it and it sounds fun

**Review by ShiftyRail — CHANGES_REQUESTED** [review]
Use the modulo `%` properly instead of using a third useless var

**Comment by boy2mantwicethefam** [comment]
it's basically a shorthand for increments right?

**Comment by Exxion** [comment]
No
It's the remainder
`....a: 0 1 2 3 4 5 6 7 8 9`
`a % 3: 0 1 2 0 1 2 0 1 2 0`
`a % 5: 0 1 2 3 4 0 1 2 3 4`

**Comment by boy2mantwicethefam** [comment]
yeah that's what I mean

**Comment by boy2mantwicethefam** [comment]
...I think

**Comment by Kharkov72** [comment]
too much
thats too much

**Comment by KingofHONK** [comment]
sickos

**Comment by boy2mantwicethefam** [comment]
Did some tests, % is not necessary, code is also slightly clearer, reviewed code however and saw that there's no check if it's above the threshold, will fix
Also there is only one more var for a total of two vars for movement (beside the one that is the sound playing), one to keep track of how much movement left until next sound play and the other to check if it's eligible to play a sound on movement

**Comment by Exxion** [comment]
No, modulo is still the correct method

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -458,16 +458,18 @@
...
+	var/step_sound_counter = 0 //Used to determine how many steps left until a sound is played
+	var/step_sound_threshold = 2 //Doesn't work if it's 0 or negative
 
 /obj/item/clothing/shoes/proc/step_action()
-	stepstaken++
 	if(step_sound != "" && ishuman(loc))
 		var/mob/living/carbon/human/H = loc
 		switch(H.m_intent)
 			if("run")
-				if(stepstaken % 2 == 1)
```
Comment (Intigracy) [inline code/modules/clothing/clothing.dm:469]: stepstaken was used right here lol

**Comment by Intigracy** [comment]
Adding thresholds as a var is fine, but there's no reason to not use modulo instead of reinventing the wheel

The clown shoes change should be its own thing honestly

**Comment by ShiftyRail** [comment]
Modulo *is* the correct method. There's a reason I hit the red cross instead of the more neutral option.
There's no need for a third var when this does the job perfectly well.

**Comment by Kurfursten** [comment]
This does not appear to have enough support.

**Pull request closed without merging**